code/game/machinery/vending.dm

```diff
@@ -577,12 +577,13 @@ var/global/num_vending_terminals = 1
 
 	if(stat & (BROKEN|NOPOWER))
 		return
-
-	if(seconds_electrified > 0)
-		if(shock(user, 100))
-			return
-	else if (seconds_electrified)
-		seconds_electrified = 0
+		
+	if(!user.lying && !user.incapacitated())
```

Comment:
The issue of not being able to see the vending machine's inventory as a ghost pops up though. 
As it is, ghosts can interact with it to see it but can't vend or spark it. 
it just werks